app/src/main/java/com/zulip/android/networking/AsyncGetEvents.java

```diff
@@ -459,4 +474,36 @@ private void processMessageEditParam(List<EventsBranch> messageEditLimitEvents)
             );
         }
     }
+
+    /**
+     * Update messages in database from list of {@link EventsBranch.BranchType#UPDATE_MESSAGE} type
+     * events {@link EventsBranch}.
+     *
+     * @param updateEvents list of events {@link EventsBranch.BranchType#UPDATE_MESSAGE}
+     */
+    private void processUpdateMessages(List<EventsBranch> updateEvents) {
+        for (EventsBranch event : updateEvents) {
+            UpdateMessageWrapper updateEvent = (UpdateMessageWrapper) event;
+            Message message = updateEvent.getMessage();
+            if (message != null) {
+                message.setFormattedContent(updateEvent.getFormattedContent());
+
+                // update the message in database
+                Dao<Message, Integer> messageDao = app.getDao(Message.class);
+                try {
+                    messageDao.update(message);
+                } catch (SQLException e) {
+                    ZLog.logException(e);
+                }
+            }
+        }
+
+        mActivity.runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                // notify adapter dataset changed
+                mActivity.getCurrentMessageList().getAdapter().notifyDataSetChanged();
```

Comment:
This notifyDataSetChanged() can be changed as we have the message ID therefore we can find the index of the message in the list and then notifyItemChanged(int);